sympy/physics/units/quantities.py

```diff
@@ -101,6 +101,22 @@ def get_dimensional_expr(expr):
         return 1
 
     @staticmethod
+    def _constructor_postprocessor_Mul(expr):
+        from .util import _get_dimension_of_expr
+        # expr._dimensional_expr = _get_dimension_of_expr(expr)
+        # setattr(expr, '_constructor_postprocessor_Add', Quantity._constructor_postprocessor_Add)
```

Comment:
Probably because of our use of `__slots__`. You'll need to add it to the slots, or store it in some other place.